finality-provider/cmd/fpd/daemon/daemon_commands.go

```diff
@@ -0,0 +1,393 @@
+	f.String(passphraseFlag, "", "The pass phrase used to encrypt the keys")
+	return cmd
+}
+
+func runCommandRegisterFP(cmd *cobra.Command, args []string) error {
+	fpPk, err := bbntypes.NewBIP340PubKeyFromHex(args[0])
+	if err != nil {
+		return err
+	}
+
+	flags := cmd.Flags()
+	daemonAddress, err := flags.GetString(fpdDaemonAddressFlag)
+	if err != nil {
+		return fmt.Errorf("failed to read flag %s: %w", fpdDaemonAddressFlag, err)
+	}
+
+	client, cleanUp, err := dc.NewFinalityProviderServiceGRpcClient(daemonAddress)
+	if err != nil {
+		return err
+	}
+	defer cleanUp()
+
+	passphrase, err := flags.GetString(passphraseFlag)
+	if err != nil {
+		return fmt.Errorf("failed to read flag %s: %w", passphraseFlag, err)
+	}
+
+	res, err := client.RegisterFinalityProvider(context.Background(), fpPk, passphrase)
+	if err != nil {
+		return err
+	}
+	printRespJSON(res)
+
+	return nil
+}
+
+// CommandAddFinalitySig returns the add-finality-sig command by connecting to the fpd daemon.
+func CommandAddFinalitySig() *cobra.Command {
+	var cmd = &cobra.Command{
+		Use:     "add-finality-sig [fp-pk-btc-hex] [block-height]",
+		Aliases: []string{"afs"},
+		Short:   "Send a finality signature to the consumer chain. This command should only be used for presentation/testing purposes",
+		Example: fmt.Sprintf(`fpd add-finality-sig --daemon-address %s`, defaultFpdDaemonAddress),
+		Args:    cobra.ExactArgs(2),
+		RunE:    runCommandAddFinalitySig,
+	}
+	cmd.Flags().String(fpdDaemonAddressFlag, defaultFpdDaemonAddress, "The RPC server address of fpd")
+	cmd.Flags().String(appHashFlag, defaultAppHashStr, "The last commit hash of the chain block")
+	return cmd
+}
+
+func runCommandAddFinalitySig(cmd *cobra.Command, args []string) error {
+	fpPk, err := bbntypes.NewBIP340PubKeyFromHex(args[0])
+	if err != nil {
+		return err
+	}
+	blkHeight, err := strconv.ParseUint(args[1], 10, 64)
+	if err != nil {
+		return err
+	}
+
+	flags := cmd.Flags()
+	daemonAddress, err := flags.GetString(fpdDaemonAddressFlag)
+	if err != nil {
+		return fmt.Errorf("failed to read flag %s: %w", fpdDaemonAddressFlag, err)
+	}
+
+	appHashHex, err := flags.GetString(appHashFlag)
+	if err != nil {
+		return fmt.Errorf("failed to read flag %s: %w", appHashFlag, err)
+	}
+
+	client, cleanUp, err := dc.NewFinalityProviderServiceGRpcClient(daemonAddress)
+	if err != nil {
+		return err
+	}
+	defer cleanUp()
+
+	appHash, err := hex.DecodeString(appHashHex)
+	if err != nil {
+		return err
+	}
+
+	res, err := client.AddFinalitySignature(context.Background(), fpPk.MarshalHex(), blkHeight, appHash)
+	if err != nil {
+		return err
+	}
+	printRespJSON(res)
+
+	return nil
+}
+
+func printRespJSON(resp interface{}) {
+	jsonBytes, err := json.MarshalIndent(resp, "", "    ")
+	if err != nil {
+		fmt.Println("unable to decode response: ", err)
+		return
+	}
+
+	fmt.Printf("%s\n", jsonBytes)
+}
+
+func loadKeyName(ctx client.Context, cmd *cobra.Command) (string, error) {
```

Comment:
```suggestion
func loadKeyName(homeDir string;, cmd *cobra.Command) (string, error) {
```

Seems we can just pass home dir here?

Reply:
done at https://github.com/babylonchain/finality-provider/pull/493/commits/964e672eababafc37e47f381aea6c635d1273cda